pkg/operator/operator/k8s.go

```diff
@@ -669,82 +676,102 @@ func getAsyncEnvVars(api spec.API, container string, queueURL string) []kcore.En
-			kcore.EnvVar{
-				Name: "HOST_IP",
-				ValueFrom: &kcore.EnvVarSource{
-					FieldRef: &kcore.ObjectFieldSelector{
-						FieldPath: "status.hostIP",
-					},
+	envVars = append(envVars,
+		kcore.EnvVar{
+			Name:  "CORTEX_LOG_LEVEL",
+			Value: strings.ToUpper(api.Predictor.LogLevel.String()),
+		},
+		kcore.EnvVar{
+			Name: "HOST_IP",
+			ValueFrom: &kcore.EnvVarSource{
+				FieldRef: &kcore.ObjectFieldSelector{
+					FieldPath: "status.hostIP",
 				},
 			},
+		},
+		kcore.EnvVar{
+			Name:  "CORTEX_SERVING_PORT",
+			Value: DefaultPortStr,
+		},
+		kcore.EnvVar{
+			Name:  "CORTEX_CACHE_DIR",
+			Value: _specCacheDir,
+		},
+		kcore.EnvVar{
+			Name:  "CORTEX_PROJECT_DIR",
+			Value: path.Join(_emptyDirMountPath, "project"),
+		},
+		kcore.EnvVar{
+			Name:  "CORTEX_DEPENDENCIES_PIP",
+			Value: api.Predictor.Dependencies.Pip,
+		},
+		kcore.EnvVar{
+			Name:  "CORTEX_DEPENDENCIES_CONDA",
+			Value: api.Predictor.Dependencies.Conda,
+		},
+		kcore.EnvVar{
+			Name:  "CORTEX_DEPENDENCIES_SHELL",
+			Value: api.Predictor.Dependencies.Shell,
+		},
+		kcore.EnvVar{
+			Name:  "CORTEX_QUEUE_URL",
+			Value: queueURL,
+		},
+		kcore.EnvVar{
+			Name:  "CORTEX_ASYNC_WORKLOAD_PATH",
+			Value: fmt.Sprintf("%s/apis/%s/workloads", config.ClusterName(), api.Name),
+		},
+		kcore.EnvVar{
+			Name:  "CORTEX_API_SPEC",
+			Value: config.BucketPath(api.PredictorKey),
+		},
+		kcore.EnvVar{
+			Name:  "CORTEX_PROCESSES_PER_REPLICA",
+			Value: s.Int32(api.Predictor.ProcessesPerReplica),
+		},
+	)
+
+	if api.Predictor.Type == userconfig.TensorFlowPredictorType {
+		envVars = append(envVars,
 			kcore.EnvVar{
-				Name:  "CORTEX_SERVING_PORT",
-				Value: DefaultPortStr,
-			},
-			kcore.EnvVar{
-				Name:  "CORTEX_CACHE_DIR",
-				Value: _specCacheDir,
-			},
-			kcore.EnvVar{
-				Name:  "CORTEX_PROJECT_DIR",
-				Value: path.Join(_emptyDirMountPath, "project"),
-			},
-			kcore.EnvVar{
-				Name:  "CORTEX_DEPENDENCIES_PIP",
-				Value: api.Predictor.Dependencies.Pip,
-			},
-			kcore.EnvVar{
-				Name:  "CORTEX_DEPENDENCIES_CONDA",
-				Value: api.Predictor.Dependencies.Conda,
-			},
-			kcore.EnvVar{
-				Name:  "CORTEX_DEPENDENCIES_SHELL",
-				Value: api.Predictor.Dependencies.Shell,
-			},
-			kcore.EnvVar{
-				Name:  "CORTEX_QUEUE_URL",
-				Value: queueURL,
+				Name:  "CORTEX_TF_BASE_SERVING_PORT",
+				Value: _tfBaseServingPortStr,
 			},
 			kcore.EnvVar{
-				Name:  "CORTEX_ASYNC_WORKLOAD_PATH",
-				Value: fmt.Sprintf("%s/apis/%s/workloads", config.ClusterName(), api.Name),
+				Name:  "CORTEX_TF_SERVING_HOST",
+				Value: _tfServingHost,
 			},
+		)
+	}
+
+	if api.Autoscaling != nil {
+		envVars = append(envVars,
 			kcore.EnvVar{
-				Name:  "CORTEX_API_SPEC",
-				Value: config.BucketPath(api.PredictorKey),
+				Name:  "CORTEX_MAX_REPLICA_CONCURRENCY",
+				Value: s.Int64(api.Autoscaling.MaxReplicaConcurrency),
 			},
+		)
+	}
+
+	if api.Predictor.Type != userconfig.PythonPredictorType || api.Predictor.MultiModelReloading != nil {
```

Comment:
`MultiModelReloading` is not permitted for the async API anyway, so I think we can remove the 2nd check. That's because `MultiModeReloading` only belongs to the python type.